### DISCUSSION TITLE
I ran NPM i react-joyride but now what?

### DISCUSSION THREAD

**Discussion by JimTechWriter** [gilbarbara/react-joyride#919] [General]
I don't understand how to get started after running npm i react-joyride. Could someone help me out?

**Comment by gilbarbara** [comment]
Hey @JimTechWriter 

There's a Get Started block in the [README](https://github.com/gilbarbara/react-joyride#getting-started). Isn't that enough?

**Comment by jimgilliam** [comment]
No, @gilbarbara, it's not... at least not for a newbie like myself. 😜 I did find this great tutorial though: https://www.telerik.com/blogs/how-to-create-interactive-guided-tours-web-applications-react-joyride, and it really helps explain most of the entire process.

**Comment by gilbarbara** [comment]
Alright, good luck!

**Comment by gradaom** [comment]
Of course! I can help you with getting started after running `npm install react-joyride`.

After successfully installing the `react-joyride` package, you can follow these steps to start using it:

1. Import the necessary components:
   
   In the file where you want to use `react-joyride`, import the following components:

   ```jsx
   import { Joyride, Step } from 'react-joyride';
   ```

2. Set up initial state for the tour:
   
   Inside your component, set up an initial state that includes the tour steps and any other options you want to configure for the tour. For example:

   ```jsx
   constructor(props) {
     super(props);
     this.state = {
       steps: [
         {
           target: '.my-element', // Target element selector or DOM node
           content: 'This is my element!', // Content for the step
           placement: 'bottom' // Placement of the tooltip (top, bottom, left, right)
         },
         // Add more steps as needed
       ],
       runTour: true // Flag to control whether the tour is active or not
     };
   }
   ```

3. Implement the `<Joyride>` component:

   Add the `<Joyride>` component to your JSX, providing it with the necessary props:

   ```jsx
   render() {
     return (
       <div>
         {/* Your other components */}
         <Joyride
           continuous // Whether to keep looping the tour when the end is reached
           run={this.state.runTour} // Whether the tour should be running or not
           scrollToFirstStep // Whether to scroll to the first step when the tour starts
           showSkipButton // Whether to show a skip button for the user to skip the tour
           steps={this.state.steps} // The array of tour steps
         />
       </div>
     );
   }
   ```

4. Customize the styling:

   By default, `react-joyride` provides minimal styling for the tour steps and tooltips. If you need to customize the styling, you can create your own CSS rules or use a CSS-in-JS solution like styled-components.

   You can also pass additional props to the `<Joyride>` component for further customization. Refer to the `react-joyride` documentation for a complete list of available props and their descriptions.

5. Start the tour:

   To start the tour, you can update the `runTour` state variable to `true`. This can be triggered by a user action such as clicking a "Start Tour" button or based on certain conditions in your application.

   ```jsx
   startTour() {
     this.setState({ runTour: true });
   }
   ```

   Make sure that the `startTour()` function is called when you want to initiate the tour.

And that's it! With these steps, you should be able to get started with `react-joyride`. Feel free to explore the `react-joyride` documentation for more advanced usage and configuration options.

If you have any specific questions or need assistance with a particular step, feel free to ask!
